libraries/cyclestreets-view/src/main/java/net/cyclestreets/views/overlay/LiveRideOverlay.java

```diff
@@ -172,15 +238,123 @@ private String speed() {
     return formatter_.speed(location.getSpeed());
   }
 
-  private String distanceUntilTurn() {
+  private int distanceUntilTurn() {
+    final GeoPoint whereIam = whereIAm();
+    final int fromEnd;
+    if (whereIam == null) {
+       fromEnd = Integer.MAX_VALUE;}
```

Comment:
Three things here:
1. Why are you using `Integer.MAX_VALUE` as a magic number, instead of the (arguably more logical) `null` value to indicate a lack of location fix?  (You would obviously need to change the return type to be an `Integer` object rather than an `int` primitive.)
2. Formatting.  Yuck!
3. Why not just return from the if branch; do away with the else branch, and return inline instead of having a `fromEnd` variable?